pkg/events/derive/net_tcp.go

```diff
@@ -0,0 +1,106 @@
+package derive
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/aquasecurity/tracee/pkg/dnscache"
+	"github.com/aquasecurity/tracee/pkg/errfmt"
+	"github.com/aquasecurity/tracee/pkg/events"
+	"github.com/aquasecurity/tracee/pkg/events/parse"
+	"github.com/aquasecurity/tracee/pkg/logger"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+func NetTCPConnect(cache *dnscache.DNSCache) DeriveFunction {
+	return deriveSingleEvent(events.NetTCPConnect, deriveNetTCPConnectArgs(cache))
+}
+
+func deriveNetTCPConnectArgs(cache *dnscache.DNSCache) deriveArgsFunction {
```

Comment:
Since I'm going to add "flow oriented events" (NetTCPFlowStart NetTCPFlowEnd or something like that) for the "new events", I think this one should be TCP only. For that to happen, I would have to pass if the socket is SOCK_STREAM or SOCK_DGRAM, WDYT ?